scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +171,373 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
+
+        p_mid = (p[:-1] + p[1:])/2
+
+    so the mesh of quantiles is refined as needed to reduce the maximum
+    "u-error"::
+
+        u_error = np.max(np.abs(dist.cdf(H(p_mid)) - p_mid))
+
+    below the specified tolerance `tol`. Refinement stops when the required
+    tolerance is achieved or when the number of mesh intervals after the next
+    refinement could exceed the maximum allowed number `max_intervals`.
+
+    The approximation will not be accurate for the extreme tails of the
+    distribution (e.g. ``H(1e-17)``, ``H(1)``).
+
+    References
+    ----------
+    .. [1] Hörmann, Wolfgang, and Josef Leydold. "Continuous random variate
+           generation by fast numerical inversion." ACM Transactions on
+           Modeling and Computer Simulation (TOMACS) 13.4 (2003): 347-362.
+
+    Examples
+    --------
+    For some distributions, ``dist.ppf`` are ``dist.rvs`` are quite slow.
+
+    >>> import numpy as np
+    >>> from scipy.stats import genexpon
+    >>> from timeit import timeit
+    >>> time_once = lambda f: f"{timeit(f, number=1)*1000:.6} ms"
+    >>> dist = genexpon(9, 16, 3)  # freeze the distribution
+    >>> p = np.linspace(0.01, 0.99, 99)  # percentiles from 1% to 99%
+    >>> time_once(lambda: dist.ppf(p))
+    '154.565 ms'  # may vary
+
+    >>> time_once(lambda: dist.rvs(size=100))
+    '148.979 ms'  # may vary
+
+    The `FastNumericalInverse` has a method that approximates ``dist.ppf``.
+
+    >>> from scipy.stats import FastNumericalInverse
+    >>> fni = FastNumericalInverse(dist)
+    >>> np.allclose(fni.ppf(p), dist.ppf(p))
+    True
+
+    In some cases, it is faster to both generate the fast numerical inverse
+    and use it than to call ``dist.ppf``.
+
+    >>> def time_me():
+    ...     fni = FastNumericalInverse(dist)
+    ...     fni.ppf(p)
+    >>> time_once(time_me)
+    '11.9222 ms'  # may vary
+
+    After generating the fast numerical inverse, subsequent calls to its
+    methods are much faster.
+    >>> time_once(lambda: fni.ppf(p))
+    '0.0819 ms'  # may vary
+
+    The fast numerical inverse can also be used to generate random variates
+    using inverse transform sampling.
+
+    >>> time_once(lambda: fni.rvs(size=100))
+    '0.0911 ms'  # may vary
+
+    Depending on the implementation of the distribution's random sampling
+    method, the random variates generated may be nearly identical, given
+    the same random state.
+
+    >>> # `seed` ensures identical random streams are used by each `rvs` method
+    >>> seed = 500072020
+    >>> rvs1 = dist.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> rvs2 = fni.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> np.allclose(rvs1, rvs2)
+    True
+
+    """
+
+    def __init__(self, dist, tol=1e-12, max_intervals=100000):
+        H, eu, intervals = _fast_numerical_inverse(dist, tol, max_intervals)
+        self.H = H
+        self.midpoint_error = eu
+        self.intervals = intervals
```

Comment:
These are not used by any methods, is there a use case to keep them? Or it's just to provide some general information? If yes, maybe implementing `__str__` (or `__repr__`) could be a nice to have.

Reply:
Huh. Well, they're not public API either, so no real reason for them. Conceivably, they are things an advanced user might want to know, but unless requested, i wouldn't expose them, so i suppose they can be removed.

Reply:
Ok, but then they can stay if you say someone might want to know. There is no harm storing them.

Reply:
Actually they are public. I guess I intended to just return them like the minimizers return iteration count, etc - implementation things, but meaningful to those who know how the algorithm works.